### ISSUE TITLE
[data grid] Autocomplete edit cell requires 2 clicks to exit edit mode

### ISSUE THREAD

**Issue by Edelf** [mui/mui-x#5506]
### Order ID 💳

28950

### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### The problem in depth 🔍

With a grid in row edit mode and a cell with the autocomplete component, it takes two clicks to exit the edit mode and commit those changes.

How to reproduce:
1) open https://codesandbox.io/s/fullfeaturedcrudgrid-demo-mui-x-forked-epnq08?file=/demo.tsx
2) select a row and edit mode
3a) if the user changes the "Name" cell and clicks once outside of the grid, it commits and goes into view mode
3b) if the user changes the "Country" cell, it takes two clicks outside the grid in order to commit and set it to view mode.

In the example above, removing the following lines resolves the click issue, but creates another problem for keyboard users where on pressing Enter to select an option it will automatically commit and exit edit mode.
`onClose={(ev) => {
        ev.preventDefault();
        ev.stopPropagation();
      }}`

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by Edelf** [comment]
@m4theushw thank you for your reply.

What I'm trying to achieve is making the "end of edit mode" consistent across cells.
In the following video you can see that it requires 1 click outside of the grid to exit the edit mode on the "Name" cell, but requires 2 clicks to exit on the "Country" cell.
https://user-images.githubusercontent.com/9094216/179425113-309360c2-081a-4f5d-84a4-fc5e2d9fa417.mp4

I don't want to exit as soon as the user selects an option, since he may want to tab to other cells. 

In summary, I want the autocomplete cell to behave exactly the same as other cells, where clicking once outside of the row or tabbing out of the row will commit and exit edit mode.

**Comment by m4theushw** [comment]
> Do you mean like this?

Yes, any call to `event.stopPropagation()` breaks the grid.

> If so, than it "breaks" the keyboard consistency where when the user presses enter to select an option it will exit edit mode:

Check the CodeSandbox I added in https://github.com/mui/mui-x/issues/5506#issuecomment-1185868527 The ideal way to disable <kbd>Enter</kbd> is using `onRowEditStop` and `event.defaultMuiPrevented = true`.

**Comment by Edelf** [comment]
> Check the CodeSandbox I added in https://github.com/mui/mui-x/issues/5506#issuecomment-1185868527 The ideal way to disable Enter is using onRowEditStop and event.defaultMuiPrevented = true.

That will disable the "Enter to commit" behavior all together which is not what I need.
I need the user to be able to select an option using the keyboard without exiting the edit mode, like so:
![firefox_5lHzAL0SjA](https://user-images.githubusercontent.com/9094216/179791223-4355e458-5e40-49b7-bab6-e24176aa06e2.gif)
But if he presses `ENTER` again after selecting an option, it will commit like any other cell.

**Comment by m4theushw** [comment]
Please check if https://codesandbox.io/s/fullfeaturedcrudgrid-demo-mui-x-forked-r3j7h4?file=/demo.tsx works. The idea is to disable the default behavior of <kbd>Enter</kbd>, then manually call `apiRef.current.stopRowEditMode()` if <kbd>Enter</kbd> was pressed while the autocomplete was closed.

**Comment by Edelf** [comment]
It worked! 
Thank you very much!
